xtask/src/main.rs

```diff
@@ -198,7 +228,320 @@ fn make_taplo_cmd(fix: bool) -> StdCommand {
     cmd
 }
 
+/// Validates a project name according to Cargo's naming conventions.
+///
+/// Adapted from Cargo's [`restricted_names`] validation.
+///
+/// [`restricted_names`]: https://github.com/rust-lang/cargo/blob/master/crates/cargo-util-schemas/src/restricted_names.rs
+///
+/// See also: <https://doc.rust-lang.org/cargo/reference/manifest.html#the-name-field>
+fn parse_project_name(name: &str) -> Result<String, String> {
+    let name = name.trim();
+
+    if name.is_empty() {
+        return Err("project name cannot be empty".into());
+    }
+
+    let mut chars = name.chars();
+    if let Some(ch) = chars.next() {
+        if ch.is_ascii_digit() {
+            return Err(format!("the name cannot start with a digit: '{}'", ch));
+        }
+        if !(ch.is_ascii_alphabetic() || ch == '_') {
+            return Err(format!(
+                "the first character must be a letter or `_`, found: '{}'",
+                ch
+            ));
+        }
+    }
+
+    for ch in chars {
+        if !(ch.is_ascii_alphanumeric() || ch == '-' || ch == '_') {
+            return Err(format!(
+                "invalid character '{}': only letters, numbers, `-`, or `_` are allowed",
+                ch
+            ));
+        }
+    }
+
+    Ok(name.to_owned())
+}
+
+fn parse_github_account(account_name: &str) -> Result<String, String> {
+    let account_name = account_name.trim();
+    if account_name.is_empty() {
+        return Err("GitHub account name cannot be empty".into());
+    }
+    Ok(account_name.to_owned())
+}
+
+fn check_project_root() -> Result<(), String> {
+    if !Path::new("Cargo.toml").exists() || !Path::new("xtask").is_dir() {
+        return Err("This command must be run from the project root directory".into());
+    }
+
+    if !Path::new("template").is_dir() {
+        return Err("The 'template' directory not found. \
+             This project may have already been bootstrapped."
+            .into());
+    }
+    Ok(())
+}
+
+fn prompt_input(prompt: &str) -> String {
+    print!("{}: ", prompt);
+    stdout().flush().unwrap();
+    let mut input = String::new();
+    stdin().read_line(&mut input).unwrap();
+    input.trim().to_owned()
+}
```

Comment:
For prompt interaction, you can make use of https://github.com/console-rs/dialoguer. It should simplify many code here.